ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -683,15 +683,19 @@ void doPhase1QBExpr(ASTNode ast, QBExpr qbexpr, String id, String alias, boolean
         qbexpr.setOpcode(QBExpr.Opcode.UNION);
         break;
       case HiveParser.TOK_INTERSECTALL:
+        queryProperties.setHasIntersect(true);
         qbexpr.setOpcode(QBExpr.Opcode.INTERSECTALL);
         break;
       case HiveParser.TOK_INTERSECTDISTINCT:
+        queryProperties.setHasIntersect(true);
         qbexpr.setOpcode(QBExpr.Opcode.INTERSECT);
         break;
       case HiveParser.TOK_EXCEPTALL:
+        queryProperties.setHasExcept(true);
         qbexpr.setOpcode(QBExpr.Opcode.EXCEPTALL);
         break;
       case HiveParser.TOK_EXCEPTDISTINCT:
+        queryProperties.setHasExcept(true);
```

Comment:
Can you please add a few unit tests for the analyzer to test that the query properties are set correctly for the different types of queries. You can get some inspiration from `TestSemanticAnalyzer` and similar classes.

Reply:
Thanks for the suggestion. I didn't have that idea. I tried to implement one, but I am not confident that this is the way. Please feel free to add any comments or suggestions to the new file.
https://github.com/apache/hive/pull/5312/commits/62e44adb4b202966479d4764e749340e939507a5